### PR TITLE
Test/payload tests

### PR DESCRIPTION
**Pull request by annasambrook**
Payload tests added to decision maker, market manager and staking skills.

### CODE REVIEW

File: packages/valory/skills/market_manager_abci/tests/__init__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2021-2024 Valory AG
```
Comment (Adamantios) [inline packages/valory/skills/market_manager_abci/tests/__init__.py:4]:
```suggestion
#   Copyright 2024 Valory AG
```

File: packages/valory/skills/staking_abci/tests/__init__.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2021-2024 Valory AG
```
Comment (Adamantios) [inline packages/valory/skills/staking_abci/tests/__init__.py:4]:
```suggestion
#   Copyright 2024 Valory AG
```

File: packages/valory/skills/market_manager_abci/tests/test_payloads.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+"""This module contains the transaction payloads for the market manager abci."""
+
+from packages.valory.skills.market_manager_abci.payloads import UpdateBetsPayload
+
+
+def test_check_stop_trading_abci_payload() -> None:
```
Comment (Adamantios) [inline packages/valory/skills/market_manager_abci/tests/test_payloads.py:24]:
```suggestion
def test_update_bets_payload() -> None:
```

File: packages/valory/skills/staking_abci/tests/test_payloads.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            {
+                "service_staking_state": 1,
+                "tx_submitter": "dummy tx submitter",
+                "tx_hash": "dummy tx hash",
+            },
+        ),
+    ],
+)
+def test_payload(
+    payload_class: Type[base_tx_payload_type], payload_kwargs: Dict
```
Comment (Adamantios) [inline packages/valory/skills/staking_abci/tests/test_payloads.py:52]: Wouldn't that work instead?

```suggestion
    payload_class: Type[BaseTxPayload], payload_kwargs: Dict
```

File: packages/valory/skills/decision_maker_abci/tests/test_payloads.py [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+                "mech_tools": "dummy mech tools",
+                "policy": "dummy policy",
+                "utilized_tools": "dummy utilized tools",
+                "selected_tool": "dummy selected tool",
+            },
+        ),
+    ],
+)
+def test_payload(
+    payload_class: Type[base_tx_payload_type], payload_kwargs: Dict
```
Comment (Adamantios) [inline packages/valory/skills/decision_maker_abci/tests/test_payloads.py:119]: Same question re the type.

**Review by Adamantios — APPROVED** [review]

**Review by Adamantios — CHANGES_REQUESTED** [review]
Please merge the new base, `v0.17.0`.
You will probably need to revert b82708062415607ee3962231bd2cf1b51b68e0be.

File: packages/valory/skills/market_manager_abci/tests/test_payloads.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    """Test `UpdateBetsPayload`."""
+
+    payload = UpdateBetsPayload(sender="sender", bets_hash="dummy bets hash")
+
+    assert payload.bets_hash == "dummy bets hash"
+    assert UpdateBetsPayload.from_json(payload.json) == payload
```
Comment (jmoreira-valory) [inline packages/valory/skills/market_manager_abci/tests/test_payloads.py:30]: why is this testing UpdateBetsPayload?

Comment (Adamantios) [reply packages/valory/skills/market_manager_abci/tests/test_payloads.py:30]: It's the only payload of this abci.

Comment (jmoreira-valory) [reply packages/valory/skills/market_manager_abci/tests/test_payloads.py:30]: test seems missplaced

Comment (Adamantios) [reply packages/valory/skills/market_manager_abci/tests/test_payloads.py:30]: Maybe the method's name is incorrect instead @annasambrook.

Comment (annasambrook) [reply packages/valory/skills/market_manager_abci/tests/test_payloads.py]: yep, it was incorrect, should be fixed now

**Comment by annasambrook** [comment]
> Please merge the new base, `v0.17.0`. You will probably need to revert [b827080](https://github.com/valory-xyz/trader/commit/b82708062415607ee3962231bd2cf1b51b68e0be).

I think I've done this correctly.

File: packages/valory/agents/trader/aea-config.yaml [hunk 7]
```diff
@@ -192,7 +192,6 @@ models:
       bet_threshold: ${int:100000000000000000}
       blacklisting_duration: ${int:3600}
       ipfs_address: ${str:https://gateway.autonolas.tech/ipfs/}
-      tools_accuracy_hash: ${str:QmR8etyW3TPFadNtNrW54vfnFqmh8vBrMARWV76EmxCZyk}
```
Comment (Adamantios) [inline packages/valory/agents/trader/aea-config.yaml:195]: Merge error.

File: packages/valory/agents/trader/aea-config.yaml [hunk 8]
```diff
@@ -258,12 +257,6 @@ models:
       bet_amount_field: ${str:collateral_amount}
       results_filename: ${str:benchmarking_results.csv}
       randomness: ${str:benchmarking_randomness}
-  acc_info_fields:
-    args:
-      tool: ${str:tool}
-      requests: ${str:total_requests}
-      accuracy: ${str:tool_accuracy}
-      sep: ${str:,}
```
Comment (Adamantios) [inline packages/valory/agents/trader/aea-config.yaml:266]: Merge error.

File: packages/valory/services/trader/service.yaml [hunk 9]
```diff
@@ -97,7 +97,6 @@ type: skill
         bet_threshold: ${BET_THRESHOLD:int:100000000000000000}
         blacklisting_duration: ${BLACKLISTING_DURATION:int:3600}
         ipfs_address: ${IPFS_ADDRESS:str:https://gateway.autonolas.tech/ipfs/}
-        tools_accuracy_hash: ${TOOLS_ACCURACY_HASH:str:QmR8etyW3TPFadNtNrW54vfnFqmh8vBrMARWV76EmxCZyk}
```
Comment (Adamantios) [inline packages/valory/services/trader/service.yaml:100]: Merge error.

File: packages/valory/services/trader/service.yaml [hunk 10]
```diff
@@ -167,13 +166,7 @@ type: skill
...
         results_filename: ${BENCHMARKING_MODE_RESULTS_FILENAME:str:benchmarking_results.csv}
         randomness: ${BENCHMARKING_MODE_RANDOMNESS:str:benchmarking_randomness}
-    acc_info_fields: &id005
-      args:
-        tool: ${ACC_INFO_FIELDS_TOOL:str:tool}
-        requests: ${ACC_INFO_FIELDS_REQUESTS:str:total_requests}
-        accuracy: ${ACC_INFO_FIELDS_ACCURACY:str:tool_accuracy}
-        sep: ${ACC_INFO_FIELDS_SEP:str:,}
-    network_subgraph: &id006
+    network_subgraph: &id005
```
Comment (Adamantios) [inline packages/valory/services/trader/service.yaml:169]: Merge error.

File: packages/valory/services/trader/service.yaml [hunk 11]
```diff
@@ -365,15 +358,14 @@ type: skill
...
-    conditional_tokens_subgraph: *id012
-    realitio_subgraph: *id013
+    network_subgraph: *id005
+    omen_subgraph: *id006
+    randomness_api: *id007
+    mech_response: *id008
+    agent_tools: *id009
+    trades_subgraph: *id010
+    conditional_tokens_subgraph: *id011
+    realitio_subgraph: *id012
```
Comment (Adamantios) [inline packages/valory/services/trader/service.yaml:368]: Merge error.

File: packages/valory/services/trader/service.yaml [hunk 12]
```diff
@@ -480,15 +472,14 @@ type: skill
...
```
Comment (Adamantios) [inline packages/valory/services/trader/service.yaml:482]: Merge error.

File: packages/valory/services/trader/service.yaml [hunk 13]
```diff
@@ -595,15 +586,14 @@ type: skill
...
```
Comment (Adamantios) [inline packages/valory/services/trader/service.yaml:596]: Merge error.

File: packages/valory/skills/trader_abci/skill.yaml [hunk 14]
```diff
@@ -271,13 +270,6 @@ models:
       results_filename: benchmarking_results.csv
       randomness: benchmarking_randomness
     class_name: BenchmarkingMode
-  acc_info_fields:
-    args:
-      tool: tool
-      requests: total_requests
-      accuracy: tool_accuracy
-      sep: ','
-    class_name: AccuracyInfoFields
```
Comment (Adamantios) [inline packages/valory/skills/trader_abci/skill.yaml:280]: Merge error.

**Review by Adamantios — CHANGES_REQUESTED** [review]

**Review by Adamantios — APPROVED** [review]

**Review by jmoreira-valory — APPROVED** [review]

**Pull request merged**